sdk/web-pubsub/web-pubsub/README.md

```diff
@@ -66,57 +52,293 @@ const key = new AzureKeyCredential("<Key>");
+A hub is a logical concept for a set of client connections. Usually you use one hub for one purpose, for example, a chat hub, or a notification hub. When a client connection connects, it connects to a hub, and during its lifetime, it belongs to that hub. Different applications can share one Azure Web PubSub service by using different hub names.
 
 ### Group
 
-Group allow broadcast messages to a subset of connections to the hub. You can add and remove users and connections as needed. A client can join multiple groups, and a group can contain multiple clients.
+A group is a subset of connections to the hub. You can add a client connection to a group, or remove the client connection from the group, anytime you want. For example, when a client joins a chat room, or when a client leaves the chat room, this chat room can be considered to be a group. A client can join multiple groups, and a group can contain multiple clients.
 
 ### User
 
 Connections to Web PubSub can belong to one user. A user might have multiple connections, for example when a single user is connected across multiple devices or multiple browser tabs.
 
 ### Message
 
-A message is either a UTF-8 encoded string or raw binary data.
+When the client is connected, it can send messages to the upstream application, or receive messages from the upstream application, through the WebSocket connection.
 
 ## Examples
 
-### Broadcast a JSON message to all users
+### Get the access token for a client to start the WebSocket connection
+
+1. Intall `ws` package for client to start the WebSocket connection
+
+```batch
+npm install ws
+```
+
+2. Update the source code to get the access token and start the WebSocket connection
 
 ```js
 const { WebPubSubServiceClient } = require("@azure/web-pubsub");
 
 const serviceClient = new WebPubSubServiceClient("<ConnectionString>", "<hubName>");
-await serviceClient.sendToAll({ message: "Hello world!" });
+
+// Get the access token for the WebSocket client connection to use
+let token = await serviceClient.getClientAccessToken();
+
+// Or get the access token and assign the client a userId
+token = await serviceClient.getClientAccessToken({ userId: "user1" });
+
+// The WebSocket client connects using the full URL containing the access token:
+const WebSocket = require("ws");
+let ws = new WebSocket(token.url);
+ws.on("open", () => console.log("connected"));
+ws.on("message", (data) => console.log(data));
+```
+
+3. Get client connection `connectionId` when the client is using subprotocol `json.webpubsub.azure.v1`:
+
+An easy way to get client connection `connectionId` is using WebSocket subprotocol `json.webpubsub.azure.v1` to connect to the service. When using this subprotocol, the client always receives JSON payload, and it receives a `connected` response JSON message when it is connected:
```

Comment:
I feel like "subprotocol" needs a concept entry? Also considering the subprotocol changes how clients send messages to the service, perhaps it is good to have the readme show everything not using the subprotocol, with a link to the subprotocol docs when there's a better way if you're using it?

Reply:
Sounds cool, move subprotocol to the samples repo